generator/mavgen.py

```diff
@@ -82,6 +83,85 @@ def mavgen(opts, args):
             print("WARNING: Unable to load XML validator libraries. XML validation will not be performed", file=sys.stderr)
             opts.validate = False
 
+    def find_unique_items(dict1, dict2):
+        """
+        Finds items present the second dictionaries and returns them as a list.
+
+        Args:
+            dict1: The first dictionary.
+            dict2: The second dictionary.
+
+        Returns:
+            A list containing items (values) that are present in only one of the dictionaries.
+        """
+        keys1 = set(dict1.keys())
+        keys2 = set(dict2.keys())
+
+        #unique_to_dict1 = [dict1[key] for key in keys1 - keys2]
+        unique_to_dict2 = [dict2[key] for key in keys2 - keys1]
+
+        return unique_to_dict2
+
+    def merge_enums(enums, included_enums):
+        '''Merge inc_enums into enums (combining duplicate enums based on name).'''
+        #print("merge %s into %s" % (ix.filename, x.filename) )
+        if not included_enums:
+            # nothing to merge
+            return
+        if not enums:
+            return
+
+        # Merge1
+        enums_dict = {an_enum.name: an_enum for an_enum in enums}
+
+        for inc_enum in included_enums:
+            if inc_enum.name in enums_dict:
+
+                an_enum = enums_dict[inc_enum.name]
+                enum_entry_dict = {an_enum_entry.name: an_enum_entry for an_enum_entry in an_enum.entry}
+                inc_enum_entry_dict = {an_enum_entry.name: an_enum_entry for an_enum_entry in inc_enum.entry}
+                only_inherited_new = find_unique_items(enum_entry_dict, inc_enum_entry_dict)
+                #print(f"Adding length enum: {len(only_inherited_new)}")
+                if len(only_inherited_new) > 0: # Only merge things from lower level
+                    #print("Merging enum: %s" % inc_enum.name)
+                    an_enum.merged = True
+                    an_enum.entry.extend(only_inherited_new)
+
+                    # sort by value
+                    an_enum.entry = sorted(an_enum.entry,
+                                       key=operator.attrgetter('value'),
+                                       reverse=False)
```

Comment:
Need to think about this bit a bit more.

This marks an enum as merged if any items are copied into it from a lower level.  That's cool, but the enum object will be the same the one in the lower list, so this will set the object in the lower list as merged, even if it isn't.

In practise this won't break anything, and with our current setup wouldn't change anything.
But we should be robust.